analyticsdataserver/views.py

```diff
@@ -71,36 +71,55 @@ class HealthView(APIView):
 
    """
     permission_classes = (permissions.AllowAny,)
-
-    def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
-        OK = 'OK'
-        UNAVAILABLE = 'UNAVAILABLE'
-
-        overall_status = UNAVAILABLE
-        db_conn_status = UNAVAILABLE
-
+    STATUS_OK = 'OK'
+    STATUS_UNAVAILABLE = 'UNAVAILABLE'
+
+    def _get_connection_status(self, connection_name):
+        """
+        With the passed in database name, try to make connection to the database.
+        If database is able to connect successfully, return 'OK' status.
+        Otherwise, return 'UNAVAILABLE' status
+        """
+        db_status = self.STATUS_UNAVAILABLE
         try:
-            connection_name = getattr(settings, 'ANALYTICS_DATABASE', 'default')
             cursor = connections[connection_name].cursor()
             try:
                 cursor.execute("SELECT 1")
                 cursor.fetchone()
-
-                overall_status = OK
-                db_conn_status = OK
+                db_status = self.STATUS_OK
             finally:
                 cursor.close()
         except Exception:  # pylint: disable=broad-except
             pass
+        return db_status
+
+    def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        overall_status = self.STATUS_UNAVAILABLE
+
+        # First try the default database.
+        # The default database hosts user and auth info. Without it, no API would function
+
+        default_db_status = self._get_connection_status('default')
+        analytics_db_status = self._get_connection_status(
+            getattr(settings, 'ANALYTICS_DATABASE')
+        )
+        analytics_v1_db_status = self._get_connection_status(
```

Comment:
it should probably only do this if the system cares about the v1 DB, i.e. settings.DATA_API_V1_ENABLED = True

Reply:
@ashultz0 There is no such setting `DATA_API_V1_ENABLED`. That setting is instead defined in the [Insights repo](https://github.com/openedx/edx-analytics-dashboard/search?q=DATA_API_V1_ENABLED)

Reply:
I imagine we don't want this to fail for existing systems w/ no v1 DB. Should we maybe have a different 'v1' healthcheck endpoint if we need to rely on the error response code?